README.md

```diff
@@ -1,24 +1,80 @@
-# Rust binding to [Open Quantum Safe][oqs]'s [liboqs][]
+liboqs-rust: Rust bindings for liboqs
+=====================================
 
-This crate provides bindings to [liboqs][]. It provides unsafe `ffi` bindings in the `oqs-sys` crate.
-Safe wrappers are offered via the `oqs` crate.
+[![Build status](https://github.com/dstebila/oqs-rs/workflows/Continuous%20integration/badge.svg)](https://github.com/dstebila/oqs-rs/actions?query=workflow%3A"Continuous+integration")
 
+**liboqs-rust** offers a Rust wrapper for the [Open Quantum Safe](https://openquantumsafe.org/) [liboqs](https://github.com/open-quantum-safe/liboqs/) C library, which is a C library for quantum-resistant cryptographic algorithms.
 
-## Adding new algorithms
+Pre-requisites
+--------------
 
-### KEMs:
+liboqs-rust depends on the [liboqs](https://github.com/open-quantum-safe/liboqs) C library; liboqs must first be compiled as a Linux/macOS/Windows library (i.e. using `ninja install` with `-DBUILD_SHARED_LIBS=ON` during configuration), see the specific platform building instructions below.
+
+<span style="color: red;">TODO: Check?</span>
+
+Contents
+--------
+
+This crate provides unsafe `ffi` bindings in the `oqs-sys` crate, and safe wrappers are offered via the `oqs` crate.
+
+Usage
+-----
+
+<span style="color: red;">TODO</span>
+
+Running
+-------
+
+<span style="color: red;">TODO</span>
+
+Adding new algorithms
+---------------------
+
+### KEMs
 
 1. Update the Git submodule
-1. `oqs-sys` will now update when you build again
-1. Add it to the ``implement_kems!`` macro call in ``oqs/src/kem.rs``:
-  * The structure is a name for the algorithm in CamelCase, and the name of the constant of the algorithm (``OQS_KEM_alg_...``)
+2. `oqs-sys` will now update when you build again
+3. Add it to the ``implement_kems!`` macro call in ``oqs/src/kem.rs``:
+  - The structure is a name for the algorithm in CamelCase, and the name of the constant of the algorithm (``OQS_KEM_alg_...``)
 
 ### Signature schemes:
 
 1. Update the Git submodule
-1. `oqs-sys` is now up-to-date when you build again
-1. Add it to ``implement_sigs!`` macro call in ``oqs/src/sig.rs``.
-  * The structure is a name for the algorithm in CamelCase, and the name of the constant of the algorithm (``OQS_SIG_alg_...``)
+2. `oqs-sys` is now up-to-date when you build again
+3. Add it to ``implement_sigs!`` macro call in ``oqs/src/sig.rs``.
+  - The structure is a name for the algorithm in CamelCase, and the name of the constant of the algorithm (``OQS_SIG_alg_...``)
+
+Limitations and security
+------------------------
+
+liboqs is designed for prototyping and evaluating quantum-resistant cryptography. Security of proposed quantum-resistant algorithms may rapidly change as research advances, and may ultimately be completely insecure against either classical or quantum computers.
```

Comment:
this might actually be something for a `SECURITY.md` file, see https://docs.github.com/en/github/managing-security-vulnerabilities/adding-a-security-policy-to-your-repository

Reply:
Working on a general security policy for OQS, but for now this will go in the README.